src/common/transformations/src/transformations/transpose_sinking/ts_reduction.cpp

```diff
@@ -158,98 +89,55 @@ TSReductionForward::TSReductionForward() {
-                }
-            }
-        }
-        bool squeeze_all_dims = false;
         if (!keep_dims) {
-            if (non_negative_axes.empty()) {
-                auto input_pshape = reduction->input_value(0).get_partial_shape();
-                if (input_pshape.is_static()) {
-                    for (size_t i = 0; i < input_pshape.size(); ++i) {
-                        if (input_pshape[i] == 1) {
-                            non_negative_axes.push_back(i);
-                        }
-                    }
-                    squeeze_all_dims = true;
-                } else {
-                    return false;
-                }
-            }
-            if (unsqueeze) {
-                transpose_order_values = get_updated_order_forward(new_values, transpose_order_values);
-            } else {
-                transpose_order_values = get_updated_order_backward(non_negative_axes, transpose_order_values);
-            }
+            transpose_order_values = GetOrderBeforeReduction(non_negative_axes, transpose_order_values);
         }
+        auto reversed_order_values = ReverseTransposeOrder(transpose_order_values);
+        auto new_transpose_order = Constant::create(transpose_order->get_element_type(),
+                                                    {transpose_order_values.size()},
+                                                    transpose_order_values);
 
-        if (!unsqueeze) {
-            auto reversed_order_values = ReverseTransposeOrder(transpose_order_values);
-            for (const auto& axis : non_negative_axes) {
-                new_values.push_back(reversed_order_values[axis]);
-            }
+        std::vector<size_t> new_values;
+        for (const auto& axis : non_negative_axes) {
+            new_values.push_back(reversed_order_values[axis]);
         }
 
-        auto new_transpose_order = std::make_shared<Constant>(transpose_order->get_element_type(),
-                                                              Shape{transpose_order_values.size()},
-                                                              transpose_order_values);
-        std::shared_ptr<Node> new_transpose, new_reduction;
-        if (squeeze_all_dims) {
-            new_transpose = transpose->clone_with_new_inputs({reduction->input_value(0), new_transpose_order});
-            new_reduction = reduction->clone_with_new_inputs({new_transpose, reduction->input_value(1)});
-        } else {
-            auto new_const =
-                std::make_shared<Constant>(reduction_axes->get_element_type(), reduction_axes->get_shape(), new_values);
-            new_transpose = transpose->clone_with_new_inputs({reduction->input_value(0), new_transpose_order});
-            new_reduction = reduction->clone_with_new_inputs({new_transpose, new_const});
-        }
+        auto new_const = Constant::create(reduction_axes->get_element_type(), reduction_axes->get_shape(), new_values);
```

Comment:
Are we sure that `new_values` contains the same number of elements as specified by `reduction_axes->get_shape()`?

Reply:
yes, this is a mistake, already fixed in the next PR